src/StoreApi/Utilities/CartController.php

```diff
@@ -803,9 +806,37 @@ function( $key, $package, $index ) {
 			);
 		}
 
-		return $calculate_rates ? wc()->shipping()->calculate_shipping( $packages ) : $packages;
+		$packages = $calculate_rates ? wc()->shipping()->calculate_shipping( $packages ) : $packages;
+
+		remove_filter( 'woocommerce_get_shipping_methods', [ $this, 'enable_local_pickup_without_address' ] );
+
+		return $packages;
 	}
 
+	/**
+	 * We want to make local pickup always avaiable without checking for a shipping zone or address.
+	 *
+	 * @param array $shipping_methods Package we're checking against right now.
+	 * @return array $shipping_methods Shipping methods with local pickup.
+	 */
+	public function enable_local_pickup_without_address( $shipping_methods ) {
+		$shipping_zones = \WC_Shipping_Zones::get_zones( 'admin' );
+		$worldwide_zone = new \WC_Shipping_Zone( 0 );
```

Comment:
We also need to fetch shipping zone 0, also known as worldwide zone, the zone that is selected when no zone is matched.

Reply:
Funny thing, the optimal thing here would be supplying a string "0" instead of int 0, saving a check. But then the next check is in reverse order and makes an extra check anyway (although it is faster than empty() ). This __construct needs some 🤍

_WC_Shipping_Zone_
```
public function __construct( $zone = null ) {
		if ( is_numeric( $zone ) && ! empty( $zone ) ) {
			$this->set_id( $zone );
		} elseif ( is_object( $zone ) ) {
			$this->set_id( $zone->zone_id );
		} elseif ( 0 === $zone || '0' === $zone ) {
			$this->set_id( 0 );
		} else {
			$this->set_object_read( true );
		}
```